apps/aecontract/src/aect_create_tx.erl

```diff
@@ -296,21 +296,25 @@ run_contract(#contract_create_tx{ nonce      =_Nonce
     Caller = owner_pubkey(Tx),
     CallStack = [], %% TODO: should we have a call stack for create_tx also
                     %% when creating a contract in a contract.
-
-    CallDef = #{ caller     => Caller
-               , contract   => ContractPubKey
-               , gas        => Gas
-               , gas_price  => GasPrice
-               , call_data  => CallData
-               , amount     => 0 %% Initial call takes no amount
-               , call_stack => CallStack
-               , code       => Code
-               , call       => Call
-               , height     => Height
-               , trees      => Trees
+    {ok, KeyBlock} = aec_chain:get_key_block_by_height(Height),
```

Comment:
Txs are processed in two places:
* Block validation in `aec_block_micro_candidate:apply_block_txs_strict` in `aec_chain_state`. This is done while no blocks are inserted in DB.
* When preparing or updating microblock candidate. This is done asynchronously to DB writes so it may lead to distinct keyblock identified in distinct txs in same microblock (different chain fork followed). Though it would not pass validation when inserting it.

So it looks ok to merge before fix https://www.pivotaltracker.com/story/show/160281213 in

Reply:
I would really like to have the keyblock hash in the microblock header and also have the beneficiary as an argument to process/check. I think it would be nice to have a functional "environment" when doing check and process with data like beneficiary, height, timestamp. Perhaps I will fix this when adding handling of timestamp to the VM.